hail/scripts/benchmark_in_batch.py

```diff
@@ -0,0 +1,293 @@
+    print(sep='\n', *[f'{path}::{name}' for path, name, _ in list_benchmarks(args.include, args.exclude)])
+
+
+def build_and_push_benchmark_image(hail_dev_image: str, artifact_uri: str, tag: str) -> str:
+    image_name = f'{artifact_uri}:{tag}'
+    with NamedTemporaryFile(mode='w+', encoding='utf-8', dir=P.curdir, prefix='Dockerfile.benchmark.') as df:
+        df.write(f"""\
+# syntax=docker/dockerfile:1.7-labs
+FROM --platform=linux/amd64 {hail_dev_image}
+
+RUN hail-apt-get-install wget
+COPY python/pytest.ini .
+COPY --exclude=**/__pycache__ --exclude=**/.pytest_cache python/benchmark benchmark
+COPY --exclude=**/__pycache__ --exclude=**/.pytest_cache python/test test
+""")
+        df.flush()
+        df.file.close()
+
+        sync_check_exec('docker', 'build', '.', '-f', df.name, '-t', image_name)
+        sync_check_exec('docker', 'push', image_name)
+        print(f'built and pushed image {image_name}')
+        return image_name
+
+
+def make_test_storage_permissions_job(b: Batch, object_prefix: str, labelled_sha: str) -> Job:
+    test_permissions = b.new_job('test permissions')
+    permissions_test_file = P.join(object_prefix, f'{labelled_sha}-permissions-test')
+    test_permissions.command(f'echo hello world > {test_permissions.permissions_test_file}')
+    b.write_output(test_permissions.permissions_test_file, permissions_test_file)
+    return test_permissions
+
+
+def make_benchmark_trial(
+    b: Batch,
+    env: Dict[str, Optional[str]],
+    path: Path,
+    benchmark_name: str,
+    trial: int,
+    deps: List[Job],  # dont reformat
+) -> Job:
+    j = b.new_job(name=f'{path}::{benchmark_name}-{trial}')
+    j.depends_on(*deps)
+    j.always_copy_output()
+
+    for varname, val in env.items():
+        if val is not None:
+            j.env(varname, val)
+
+    j.env('BENCHMARK_TRIAL_ID', str(trial))
+
+    j.command('mkdir -p benchmark-resources')
+    j.command(f'touch {j.ofile}')
+    j.command(f"""\
+python3 -m pytest {path} \
+    -Werror:::hail -Werror:::hailtop -Werror::ResourceWarning \
+    --log-cli-level=INFO \
+    -s \
+    -r A \
+    -vv \
+    --instafail \
+    --durations=50 \
+    --timeout=1800 \
+    --override-ini=tmp_path_retention_count=0 \
+    --override-ini=tmp_path_retention_policy=failed \
+    --output={j.ofile} \
+    --data-dir=benchmark-resources \
+    -k {benchmark_name}
+""")
+    return j.ofile
+
+
+def read_jsonl(p: Path):
+    logging.info(f'reading json lines from {p}.')
+    with p.open(encoding='utf-8') as r:
+        for line in r:
+            if len(line) > 1:
+                yield json.loads(line)
+
+
+def combine(output: Resource, files: List[Resource]) -> None:
+    init_logging()
+    n_files = len(files)
+    if n_files < 1:
+        raise ValueError("'combine' requires at least 1 file to merge")
+
+    logging.info(f'combining {len(files)} files')
+
+    jsonl = [line for f in files if (p := Path(f)).exists() for line in read_jsonl(p)]
```

Comment:
Since you're now going with the `touch` approach for the benchmark trial jobs, can you remove the existence check? Or is there still a scenario where the file wouldn't exist?

Reply:
true